com.unity.ml-agents.extensions/Runtime/Sensors/RigidBodyHierarchyUtil.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.MLAgents.Extensions.Sensors
+{
+
+    /// <summary>
+    /// Utility class to track a hierarchy of RigidBodies. These are assumed to have a root node,
+    /// and child nodes are connect to their parents via Joints.
+    /// </summary>
+    public class RigidBodyHierarchyUtil : HierarchyUtil
+    {
+        Rigidbody[] m_Bodies;
+
+        /// <summary>
+        /// Initialize given a root RigidBody.
+        /// </summary>
+        /// <param name="rootBody"></param>
+        public RigidBodyHierarchyUtil(Rigidbody rootBody)
```

Comment:
There are some assumptions here that the bodies don't form a loop. Might need some explicit checks for this eventually.